casper/src/main/resources/PoS.rhox

```diff
@@ -155,26 +154,64 @@ new  PoS,
               }
             } |
 
-            contract PoS (@"chargeDeploy", @deployerId, @amount, returnCh) = {
-              new vaultCh, transferCh, userCh, processPayCh,
-                  depositCh,
-                  newPendingCh in {
-                getUser!(deployerId, *userCh) |
-                runMVar!(*stateCh, *processPayCh, *returnCh) |
-                for(@state, payResultCh <- processPayCh;
-                    @userPk <- userCh) {
+            new currentDeployerData in {
+              contract PoS (@"chargeDeploy", @deployerId, @amount, return) = {
+                new depositCh in {
                   deposit!(deployerId, amount, *depositCh) |
-
-                  distributeRewards!(
-                    amount,
-                    state.get("activeValidators"),
-                    state.get("allBonds"),
-                    state.get("pendingRewards"),
-                    *newPendingCh) |
-
-                  for(@depositResult <- depositCh;
-                      @newPending <- newPendingCh) {
-                    payResultCh!(state.set("pendingRewards", newPending), depositResult)
+                  for(@depositResult <- depositCh){
+                    match depositResult {
+                      (true, _) => {
+                        currentDeployerData!((deployerId, amount)) |
+                        return!(depositResult)
+                      }
+                      (_, errorMessage) => {
+                        return!((false, "Deploy payment failed: " ++ errorMessage))
+                      }
+                    }
+                  }
+                }
+              } |
+
+              // Note: under no circumstances should this method be merged as is.
+              //       this method, along with "chargeDeploy", "closeBlock", and "slash"
+              //       must be made private, as they're only used in the protocol.
+              contract PoS (@"refundDeploy", @refundAmount, return) = {
+                for(@(deployerId, initialPaymentAmount) <- currentDeployerData){
+                  new revAddressCh in {
+                    revAddressOps!("fromDeployerId", deployerId, *revAddressCh) |
+                    for(@deployerRevAddress <- revAddressCh){
+                      new transferCh in {
```

Comment:
I don't know if it's a rule but I agree it would look better because it reduces the indentation level.

Reply:
It feels wrong to move the name to the top of the file because names should be scoped according to their use. Moving all names to the top of the file is akin to declaring all values at the top-level in Scala or something. It also makes it harder to find out where names are actually used. But, if both of you insist, I'll move the channel to the top of the file.

Reply:
I don't insist. I just wanted to mention that we have been doing it this way so far :)